vispy/visuals/tests/test_volume.py

```diff
@@ -467,4 +491,52 @@ def test_minip_cutoff():
         assert np.array_equal(rendered[40, 40], [0, 0, 0, 255])
 
 
+@requires_pyopengl()
+@requires_application()
+def test_volume_set_data_different_dtype():
+    size = (80, 80)
+    data = np.array([[[0, 127]]], dtype=np.int8)
+    left = (40, 10)
+    right = (40, 70)
+    white = (255, 255, 255, 255)
+    black = (0, 0, 0, 255)
+
+    with TestingCanvas(size=size[::-1], bgcolor="w") as c:
+        view = c.central_widget.add_view()
+        view.camera = 'arcball'
+        view.camera.fov = 0
+        view.camera.center = 0.5, 0, 0
+        view.camera.scale_factor = 2
+        volume = scene.visuals.Volume(
+            data,
+            cmap='grays',
+            clim=[0, 127],
+            parent=view.scene
+        )
+
+        render = c.render()
+        assert np.allclose(render[left], black)
+        assert np.allclose(render[right], white)
+
+        # same data as float should change nothing
+        volume.set_data(data.astype(np.float32))
+        render = c.render()
+        assert np.allclose(render[left], black)
+        assert np.allclose(render[right], white)
+
+        # something inverted, different dtype
+        new_data = np.array([[[127, 0]]], dtype=np.float16)
+        volume.set_data(new_data)
+        render = c.render()
+        assert np.allclose(render[left], white)
+        assert np.allclose(render[right], black)
+
+        # out of bounds should wrap around cleanly
+        new_data = np.array([[[0, 128]]], dtype=np.float64)
+        volume.set_data(new_data)
+        render = c.render()
+        assert np.allclose(render[left], black)
+        assert np.allclose(render[right], black)
```

Comment:
Same argument as for the ImageVisual test, shouldn't 128 get clipped to 127 and be white on the output?